frontend/src/lib/components/ExportButton/exporterLogic.ts

```diff
@@ -64,57 +64,62 @@ export const exporterLogic = kea<exporterLogicType>([
 
     listeners(({ actions, props }) => ({
         exportItem: async ({ exportFormat, exportContext, successCallback }) => {
-            lemonToast.info(`Export started...`)
-
-            const trackingProperties = {
-                export_format: exportFormat,
-                dashboard: props.dashboardId,
-                insight: props.insightId,
-                total_time_ms: 0,
-            }
-            const startTime = performance.now()
-
-            try {
-                let exportedAsset = await api.exports.create({
+            const poller = new Promise(async (resolve, reject) => {
+                const trackingProperties = {
                     export_format: exportFormat,
                     dashboard: props.dashboardId,
                     insight: props.insightId,
-                    ...(exportContext || {}),
-                })
-
-                if (!exportedAsset.id) {
-                    throw new Error('Missing export_id from response')
+                    total_time_ms: 0,
                 }
+                const startTime = performance.now()
+
+                try {
+                    let exportedAsset = await api.exports.create({
+                        export_format: exportFormat,
+                        dashboard: props.dashboardId,
+                        insight: props.insightId,
+                        ...(exportContext || {}),
+                    })
+
+                    if (!exportedAsset.id) {
+                        reject('Missing export_id from response')
+                    }
 
-                let attempts = 0
+                    let attempts = 0
 
-                while (attempts < MAX_POLL) {
-                    attempts++
+                    while (attempts < MAX_POLL) {
+                        attempts++
 
-                    if (exportedAsset.has_content) {
-                        actions.exportItemSuccess()
-                        lemonToast.success(`Export complete.`)
-                        successCallback?.()
-                        await downloadExportedAsset(exportedAsset)
+                        if (exportedAsset.has_content) {
+                            actions.exportItemSuccess()
+                            successCallback?.()
+                            await downloadExportedAsset(exportedAsset)
 
-                        trackingProperties.total_time_ms = performance.now() - startTime
-                        posthog.capture('export succeeded', trackingProperties)
+                            trackingProperties.total_time_ms = performance.now() - startTime
+                            posthog.capture('export succeeded', trackingProperties)
 
-                        return
-                    }
+                            resolve('Export complete')
+                            break
```

Comment:
Why `break` and not `return`? This will theoretically also call the `reject` after the loop.

Reply:
Well, in my mind once resolved it can't be rejected

But that linked SO answer has destroyed my mental model of Promises :) I trust nothing now :D

Reply:
Functions are still functions with their own control flow, no matter what other functions they call :)

Reply:
I just want to believe in magic 🌤️